Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/WMB.hs

```diff
@@ -6,51 +6,299 @@ module Domain.Action.UI.WMB
+  infoSrc <- QRTS.findByLocation res.startPoint >>= fromMaybeM (InternalError "start point not found")
+  infoDest <- QRTS.findByLocation res.endPoint >>= fromMaybeM (InternalError "end point not found")
+  let routeData =
+        RouteInfo
+          { code = res.code,
+            shortName = res.shortName,
+            longName = res.longName,
+            startPoint = res.startPoint,
+            endPoint = res.endPoint
+          }
+  let vhclDetails =
+        VehicleDetails
+          { number = vhclNo,
+            _type = vehicleServiceTierType
+          }
+  let srcInfo =
+        StopInfo
+          { name = infoSrc.stopName,
+            code = infoSrc.stopCode,
+            lat = Just res.startPoint.lat,
+            long = Just res.startPoint.lon
+          }
+  let dstInfo =
+        StopInfo
+          { name = infoDest.stopName,
+            code = infoDest.stopCode,
+            lat = Just res.endPoint.lat,
+            long = Just res.endPoint.lon
+          }
+  let availableRoutesList =
+        AvailableRoute
+          { routeInfo = routeData,
+            source = srcInfo,
+            destination = dstInfo,
+            vehicleDetails = vhclDetails
+          }
+  pure availableRoutesList
 
 getUiWmbAvailableRoutes ::
   ( ( Kernel.Prelude.Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person),
       Kernel.Types.Id.Id Domain.Types.Merchant.Merchant,
       Kernel.Types.Id.Id Domain.Types.MerchantOperatingCity.MerchantOperatingCity
     ) ->
     Data.Text.Text ->
-    Environment.Flow [API.Types.UI.WMB.AvailableRoutesList]
+    Environment.Flow [API.Types.UI.WMB.AvailableRoute]
   )
-getUiWmbAvailableRoutes = do error "Logic yet to be decided"
+getUiWmbAvailableRoutes (person, _, _) vhclNumberHash = do
+  checkFleetDriverAssociation person >>= \isAssociated -> unless isAssociated (throwError $ InternalError "Fleet-driver association not found")
+  let vehicleNumberHash = DbHash $ encodeUtf8 vhclNumberHash
+  result <-
+    mapM
+      ( \mapping -> do
+          vehicleNumber <- decrypt mapping.vehicleNumber
+          availableRoutes (mapping.routeCode, mapping.vehicleServiceTierType) vehicleNumber
+      )
+      =<< VRME.findAllRouteMappings vehicleNumberHash
+  pure result
 
 postUiWmbTripLink ::
   ( ( Kernel.Prelude.Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person),
       Kernel.Types.Id.Id Domain.Types.Merchant.Merchant,
       Kernel.Types.Id.Id Domain.Types.MerchantOperatingCity.MerchantOperatingCity
     ) ->
     API.Types.UI.WMB.TripLinkReq ->
-    Environment.Flow API.Types.UI.WMB.TripLinkResp
+    Environment.Flow API.Types.UI.WMB.TripTransactionDetails
   )
-postUiWmbTripLink = do error "Logic yet to be decided"
+postUiWmbTripLink (person, mer, city) obj = do
+  driverId <- fromMaybeM (InternalError "Driver ID not found") person
+  fleetDriverAssociation <- FDA.findByDriver driverId >>= fromMaybeM (InternalError "Fleet Association not found")
```

Comment:
check the comment above, we don't want to fail this API is a driver is not linked to any fleet, just link the driver to the fleet to which the vehicleHash in `TripLinkReq` belongs